drivers/overlay/overlay.go

```diff
@@ -264,23 +264,39 @@ func parseOptions(options []string) (*overlayOptions, error) {
 	return o, nil
 }
 
-func supportsOverlay() error {
+func supportsOverlay(home string, rootUID, rootGID int) error {
 	// We can try to modprobe overlay first before looking at
 	// proc/filesystems for when overlay is supported
 	exec.Command("modprobe", "overlay").Run()
 
-	f, err := os.Open("/proc/filesystems")
-	if err != nil {
-		return err
-	}
-	defer f.Close()
-
-	s := bufio.NewScanner(f)
-	for s.Scan() {
-		if s.Text() == "nodev\toverlay" {
-			return nil
+	// Try a test mount in the specific location we're looking at using.
+	layerDir, err := ioutil.TempDir(home, "compat")
+	if err == nil {
+		mergedDir := filepath.Join(layerDir, "merged")
+		upperDir := filepath.Join(layerDir, "diff")
+		workDir := filepath.Join(layerDir, "work")
+		lowerDir := filepath.Join(layerDir, "lower")
+		defer func() {
+			// Permitted to fail, since the various subdirectories
+			// can be empty or not even there, and the home might
+			// legitimately be not empty
+			_ = unix.Unmount(mergedDir, unix.MNT_DETACH)
+			_ = os.RemoveAll(layerDir)
+			_ = os.Remove(home)
```

Comment:
Why are you removing home?  That was passed in,  I don't thing the function should remove this.

Reply:
This bit runs before we "commit" to using the driver, so if the directory's empty except for the things we've just created, we try remove it, but not recursively.  If it fails, fine, but if it succeeds, then we avoid confusing `scanPriorDrivers()`.